surefire-providers/common-junit4/src/main/java/org/apache/maven/surefire/common/junit4/JUnit4RunListener.java

```diff
@@ -41,12 +43,14 @@
  *
  */
 public class JUnit4RunListener
-    extends org.junit.runner.notification.RunListener
+    extends RunListener
+    implements TestOutputReceiver
 {
-    protected final RunListener reporter;
+    protected final TestRunListener reporter;
```

Comment:
maybe also change reported -> testRunListener
and in other similar place

Reply:
The interface `TestReportListener` contains the wording `report`, pls see the next commit.